### ISSUE TITLE
Conflict with phonegap-plugin-push on Android

### ISSUE THREAD

**Issue by ahmad6242** [EddyVerbruggen/cordova-plugin-safariviewcontroller#84]
I have installed that plugin and use it on Android and IOS... it is working perfectly...
But what i observe is that my phonegap-plugin-push stop generating the Unique App ID in Android. 
After 6 hours of Debugging about that issue i found out that whenever i removed cordova-plugin-safariviewcontroller from the project phonegap-plugin-push starts generating the Unique ID for the APP. 
I dont understand what's conflicting between these two Plugins... Maybe you can find something. 

I will provide you every detail if you understand that this is some kind of issue, and you will like to help... 
Thanx 👍

**Comment by youssmak** [comment]
@ahmad6242 Try the new version 1.5.0 which includes : https://github.com/EddyVerbruggen/cordova-plugin-safariviewcontroller/pull/86

It should fix your issue

**Comment by rafalowski** [comment]
Same issue here!!! Urgent Issue!!! Must be fixed!!!

Any updates on this

**Comment by rafalowski** [comment]
Why deleting the comment.
I also noticed that push notification stopped working on Andriod

**Comment by EddyVerbruggen** [comment]
It was deleted because of the tone. I don't recall what it was and I don't care what it was. If you have anything to add that helps solving this issue please go ahead. Otherwise, subscribing to this issue or leaving a thumb at the OP's post is sufficient.

**Comment by jespertheend** [comment]
Is #86 supposed to fix this issue?
I'm using version 1.5.3 of this plugin (and phonegap-plugin-push 2.1.3) but I still seem to be having this issue. Is there something I'm doing wrong?

**Comment by adrenaline15** [comment]
I also still have the same issue and the same versions you mentioned @jespertheend... 

Even tried older versions, but still no ĺuck.. The phongap-push-plugin always throws this error at the startup: `Got Resources NotFoundException String resource ID #0x0`

**Comment by jespertheend** [comment]
@adrenaline15 I ended up removing this plugin in my build script when building for android. I think it might be related to android@6.3.0 instead of android@7.1.0, but I have no idea, this works for me. It's too bad I can't have chrome custom tabs, but I suppose the built in in app browser works ok for android.

**Comment by markphillips100** [comment]
I just started getting this issue as I've added SVC to support a more secure authentication workflow. I read in one issue comment (maybe in the push repo) that the ordering of the plugin install (into the platforms folder) may have an impact, as it seems one plugin is impacting another's files (don't ask me how).

The app's fetch.json file in the plugins folder had the push plugin nearer the top of the file than the SVC plugin, and I believe the plugins will be added to the platforms folder in the order listed.  I was suspicious so decided to remove and add the platform with the plugin order changed.

FYI I'm using Ionic 3, cordova-android@7.1.0, windows 10 and vscode.  

1. ionic cordova platform rm android
2. edit fetch.json and move cordova-plugin-push config below SVC's config
3. ionic cordova platform add android@7.1.0
4. build and debug

After doing this I had no complaint from the push plugin.

**Comment by larssn** [comment]
Can you post your fetch.json?

**Comment by markphillips100** [comment]
Sure.  I've added it below. Please note the FCM version specified for the push plugin.  This was originally required to resolve an issue using the plugin alongside a google maps sdk plugin.  I no longer use that maps plugin and instead use the JS API.  Not sure if it makes a difference in this case or not.

```
{
  "cordova-plugin-whitelist": {
    "source": {
      "type": "registry",
      "id": "cordova-plugin-whitelist@^1.3.3"
    },
    "is_top_level": true,
    "variables": {}
  },
  "cordova-plugin-device": {
    "source": {
      "type": "registry",
      "id": "cordova-plugin-device@^2.0.1"
    },
    "is_top_level": true,
    "variables": {}
  },
  "cordova-plugin-inappbrowser": {
    "source": {
      "type": "registry",
      "id": "cordova-plugin-inappbrowser@^1.6.1"
    },
    "is_top_level": true,
    "variables": {}
  },
  "cordova-plugin-splashscreen": {
    "source": {
      "type": "registry",
      "id": "cordova-plugin-splashscreen@^5.0.2"
    },
    "is_top_level": true,
    "variables": {}
  },
  "cordova-plugin-ionic-webview": {
    "source": {
      "type": "registry",
      "id": "cordova-plugin-ionic-webview@^1.1.16"
    },
    "is_top_level": true,
    "variables": {}
  },
  "cordova-plugin-ionic-keyboard": {
    "source": {
      "type": "registry",
      "id": "cordova-plugin-ionic-keyboard@^2.0.5"
    },
    "is_top_level": true,
    "variables": {}
  },
  "cordova-plugin-geolocation": {
    "source": {
      "type": "registry",
      "id": "cordova-plugin-geolocation@4.0.1"
    },
    "is_top_level": true,
    "variables": {}
  },
  "cordova-plugin-camera": {
    "source": {
      "type": "registry",
      "id": "cordova-plugin-camera@4.0.2"
    },
    "is_top_level": true,
    "variables": {}
  },
  "cordova-plugin-file": {
    "source": {
      "type": "registry",
      "id": "cordova-plugin-file@6.0.1"
    },
    "is_top_level": true,
    "variables": {}
  },
  "cordova-plugin-file-transfer": {
    "source": {
      "type": "registry",
      "id": "cordova-plugin-file-transfer"
    },
    "is_top_level": true,
    "variables": {}
  },
  "cordova-plugin-compat": {
    "source": {
      "type": "registry",
      "id": "cordova-plugin-compat@^1.0.0"
    },
    "is_top_level": false,
    "variables": {}
  },
  "cordova-universal-links-plugin": {
    "source": {
      "type": "registry",
      "id": "https://github.com/walteram/cordova-universal-links-plugin"
    },
    "is_top_level": true,
    "variables": {}
  },
  "cordova-plugin-safariviewcontroller": {
    "source": {
      "type": "registry",
      "id": "cordova-plugin-safariviewcontroller"
    },
    "is_top_level": true,
    "variables": {}
  },
  "phonegap-plugin-push": {
    "source": {
      "type": "registry",
      "id": "phonegap-plugin-push@2.1.3"
    },
    "is_top_level": true,
    "variables": {
      "FCM_VERSION": "11.8.0"
    }
  }
}
```

**Comment by larssn** [comment]
Looking at https://github.com/phonegap/phonegap-plugin-push/issues/1800 this seems to be an issue with the push plugin, and not this. So I'm gonna close this issue.

If you find anything that shows the problem can be fixed in our end, just post here, and I'll reopen.

**Comment by rdesimone** [comment]
IMO the gradle file has to be removed - at least when building with the latest tools - this works for me. cc @EddyVerbruggen 

`<framework src="build/android/SafariViewController-java18.gradle" custom="true" type="gradleReference" />`

**Comment by jespertheend** [comment]
@rdesimone are you saying that line should be removed from this plugins `plugin.xml`. I tried that but that didn't work.

**Comment by rdesimone** [comment]
Yes, I removed it in my fork! After removing and adding the plugin, check in Android Studio `build.gradle (Module: app)` that it really does not appear anymore in the `// PLUGIN GRADLE EXTENSIONS START` section.

Check also that you don't mix dependencies version. Set for all the same version, e.g:

```
dependencies {
    implementation fileTree(include: '*.jar', dir: 'libs')
    // SUB-PROJECT DEPENDENCIES START
    implementation(project(path: ":CordovaLib"))
    compile "com.android.support:appcompat-v7:27.1.0"
    compile "com.android.support:support-v4:27.1.0"
    compile "com.google.firebase:firebase-core:11.8.0"
    compile "com.android.support:support-v13:27.1.0"
    compile "me.leolin:ShortcutBadger:1.1.17@aar"
    compile "com.google.firebase:firebase-messaging:11.8.0"
    compile "com.android.support:customtabs:27.1.0"
    compile "com.google.android.gms:play-services-maps:11.8.0"
    compile "com.google.android.gms:play-services-location:11.8.0"
    compile "com.android.support:appcompat-v7:27.1.0"
    // SUB-PROJECT DEPENDENCIES END
}
````

**Comment by adrenaline15** [comment]
It seems to work if I update my cordova-android version to 7.1.0.

Now I can use the _safariviewcontroller-plugin_ and the _phonegap-push-plugin_ together!

Thanks a lot @markphillips100  @jespertheend

**Comment by jespertheend** [comment]
Hi @rdesimone 
Are you using cordova-android@7.1.0? I tried your fork but I'm getting this error: `diamond operator is not supported in -source 1.6`. I suppose that's because I'm still using android@6.3.0.
Or did you manage to solve this in some other way?

**Comment by rdesimone** [comment]
Yes, I am using the latest tools as I wrote in my first comment. 

`cordova 8.0.0`
`android 7.1.0` 
`Android Studio 3.0.1`
`phonegap-plugin-push 2.2.1 "PushPlugin"`

My fork is basically the same but it sets `"com.android.support:customtabs:27.1.0" and it fixes the mentioned problem to work together with the push plugin.

**Comment by eduboxgithub** [comment]
I'm using this versions:

```
Cordova 7.1.0
Android Plataform 6.3.0
My Push plugin version is 2.1.0
```

And my plugin list is:

```
code-push 2.0.4 "CodePushAcquisition"
cordova-plugin-actionsheet 2.3.3 "ActionSheet"
cordova-plugin-android-permissions 1.0.0 "Permissions"
cordova-plugin-calendar 5.1.2 "Calendar"
cordova-plugin-camera 4.0.3 "Camera"
cordova-plugin-code-push 1.11.11 "CodePush"
cordova-plugin-compat 1.2.0 "Compat"
cordova-plugin-configsedubox 0.0.1 "Plugin Apps Edubox"
cordova-plugin-device 2.0.2 "Device"
cordova-plugin-dialogs 2.0.1 "Notification"
cordova-plugin-facebook4 2.1.0 "Facebook Connect"
cordova-plugin-file 4.3.3 "File"
cordova-plugin-file-transfer 1.6.3 "File Transfer"
cordova-plugin-geolocation 4.0.1 "Geolocation"
cordova-plugin-inappbrowser 3.0.0 "InAppBrowser"
cordova-plugin-ionic-webview 1.2.1 "cordova-plugin-ionic-webview"
cordova-plugin-overview 0.1.2 "Overview"
cordova-plugin-splashscreen 5.0.2 "Splashscreen"
cordova-plugin-statusbar 2.4.2 "StatusBar"
cordova-plugin-whitelist 1.3.3 "Whitelist"
cordova-plugin-x-socialsharing 5.4.0 "SocialSharing"
cordova-plugin-zip 3.1.0 "cordova-plugin-zip"
es6-promise-plugin 4.2.2 "Promise"
ionic-plugin-keyboard 2.2.1 "Keyboard"
phonegap-plugin-push 2.1.0 "PushPlugin"
uk.co.workingedge.phonegap.plugin.launchnavigator 4.2.1 "Launch Navigator"
```

I'm also getting the error right after the init call: 

```
Error with Push plugin Error: String resource ID #0x0
    at fail (push.js:62)
    at Object.callbackFromNative (cordova.js:296)
    at <anonymous>:1:9
```

I have the configuration files (google-services.json) in the root project and the <resource-file> correctly set in the config.xml.

**Comment by hasib2cool** [comment]
> IMO the gradle file has to be removed - at least when building with the latest tools - this works for me. cc @EddyVerbruggen
> 
> `<framework src="build/android/SafariViewController-java18.gradle" custom="true" type="gradleReference" />`

IT is still not generating the app it is showing me the error 👍 
Error compiling Cordova plugin: org/chromium/customtabsclient/shared/CustomTabsHelper.java:77: error: diamond operator is not supported in -source 1.6.

**Comment by uosl** [comment]
I tried several solutions (using cordova-android@7.1.0 and using cdvPluginPostBuildExtras in `build-extras.gradle`, as mentioned in https://github.com/phonegap/phonegap-plugin-push/issues/1800) but the only thing that worked consistently to fix `Error: String resource ID #0x0` for me was to edit `platforms/android/project.properties`:
```
cordova.gradle.include.3=cordova-plugin-safariviewcontroller/myapp-SafariViewController-java18.gradle
cordova.gradle.include.1=cordova-support-google-services/myapp-build.gradle
cordova.gradle.include.2=phonegap-plugin-multidex/myapp-multidex.gradle
```
The number after *include* doesn't matter, as long as the line with `cordova-plugin-safariviewcontroller` is above the *include* line for `cordova-support-google-services` (a dependency of phonegap-plugin-push). I guess it happens because both plugins use `ext.postBuildExtras` in their gradle file.

I hope someone familiar with gradle files will be able to tell where and how this conflict should be resolved, so that the order they are specified in that file doesn't matter.

**Comment by uosl** [comment]
I looked into this some more and here's what I found:

According to https://github.com/chemerisuk/cordova-support-google-services/issues/9, cordova plugins should use `ext.cdvPluginPostBuildExtras` instead of `ext.postBuildExtras`, which neither [cordova-plugin-safariviewcontroller](https://github.com/EddyVerbruggen/cordova-plugin-safariviewcontroller/blob/f46472865acd431e6775cd85ba39437465b059c8/build/android/SafariViewController-java18.gradle#L1) and [cordova-support-google-services](https://github.com/chemerisuk/cordova-support-google-services/blob/83949a7afe06c5537c801fa16a1af936e2a06ddb/build.gradle#L15) currently does, causing them to overwrite each other.

A simple temporary fix I used is to add this to `build-extras.gradle`. (remember to replace *myapp*)
```
apply from: "../cordova-support-google-services/myapp-build.gradle"
```
Then use *resource-file* in `config.xml` to copy it into the platforms/android directory.
```
    <platform name="android">
        <resource-file src="build-extras.gradle" target="build-extras.gradle" />
    </platform>
```
Depending on your setup and plugins, this might not be enough and in the worst case you'll have to fork (or cordova plugin add a local directory) one or both of the problem plugins and add in `ext.cdvPluginPostBuildExtras` yourself.
